pkg/build/builder/cmd/scmauth/scmauth.go

```diff
@@ -0,0 +1,25 @@
+package scmauth
```

Comment:
I'm open for naming propositions, I suck at it :wink: As for location of that package, since this is tightly connected with builds, I've decided to put it here. As for externalization, I'm wondering how useful these auth methods will be for other SCM. I know mercurial supports SSH keys, but the setup might be different. If so then packages could be named then "pkg/util/scm/git/auth/"